feat(MultipleLanguage): add app support to multiple languages

anderson-olive opened the pull request:
# Description

This PR add the support for multiple languages in our app using the recommended [next-i18next package](https://github.com/i18next/next-i18next), all the documentation have been updated with guides how to use.
In order to use the translation, the **serverSideTranslations HOC** should be added to all pages inside the /pages folder in the getStaticProps function. For example:

```
export const getStaticProps: GetStaticProps = async (context) => {
  let { locale } = context;
  return {
    props: {
      ...(await serverSideTranslations(locale ?? 'en')),
    },
  };
};
```

Using getStaticProps, we can access the locale parameter from the navigation, and we can pass the JSON translations to our page.

## Changes included:

- Setup next-i18next
- Add a locales folder with translations (public/locales).
- Update pages to use the "useTranslations" hook.
- Add a new icon, "LanguageIcon."
- Add a new component, "LanguageSwitcher," to manage the current language.
- Update the 404 page UI.
- Update documentation to add multilingual information.
- Create a suggestion for the pull_request_template.
- Add the Next_LOCALE cookie to persist user selection.
- Add a new package, "nookies," to manage cookies on the server and client sides.
- Add a new package "next-i8next" to manage internationalization.

## Demo

![demo-multiple-language](https://github.com/nandoramos/youtube-gpt/assets/106548895/6b072c13-c240-43a3-91f1-d542556cb88e)


## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

anderson-olive committed:
feat(MultipleLanguage): add app support to multiple languages

- Setup next-i18next
- Add locales folder with translations
- Update pages to use the "useTranslations" hook
- Add new icon "LanguageIcon"
- Add a new component "LanguageSwitcher" to manage the current language
- Update 404 page UI

anderson-olive committed:
doc: update the project documentation

- Update documentation to add multilingual information
- Create a suggestion for pull_request_template

anderson-olive committed:
doc: update logo size

- Update logo size

anderson-olive committed:
docs: add project description

- Add project description

nandoramos commented on youtube-gpt-client/src/components/header/index.tsx:
```diff
@@ -1,20 +1,29 @@
-        <Link href="/how-it-works">How it works</Link>
-        <Link href="/team">Team</Link>
+      <Flex gap="20px" fontSize="20px">
+        <Link href="/how-it-works" locale={i18n.language}>
```
is this locale={i18n.language} needed? I see it repeated and wondering if it can be solved in some global config instead of in every use?

anderson-olive replied:
I've updated the code to store the [NEXT_LOCALE cookie](https://nextjs.org/docs/pages/building-your-application/routing/internationalization#leveraging-the-next_locale-cookie). This way, we don't need to set all links and work in the entire application. I'm using [nookies](https://github.com/maticzav/nookies) to set the value because we can retrieve the cookies from the server side and client side of the application with ease.

anderson-olive committed:
feat(MultipleLanguage): add NEXT_LOCALE cookie to persist

- Add NEXT_LOCALE cookie to persist the locale using nookie
- Code clean up
- Add demo section to pull_request template

nandoramos reviewed (approved):
👍
